network/tls_wrapper.go

```diff
@@ -47,35 +48,30 @@ type TLSConnectionWrapper struct {
 // ErrEmptyTLSConfig if not TLS config found
 var ErrEmptyTLSConfig = errors.New("empty TLS config")
 
-// NewTLSConnectionWrapper returns new TLSConnectionWrapper
-func NewTLSConnectionWrapper(clientID []byte, config *tls.Config) (*TLSConnectionWrapper, error) {
-	return &TLSConnectionWrapper{config: config, clientID: clientID}, nil
+var (
+	tlsCA         string
+	tlsKey        string
+	tlsCert       string
+	tlsAuthType   int
+	tlsServerName string
+)
+
+// RegisterTLSBaseArgs register CLI args tls_ca|tls_key|tls_cert|tls_auth which allow to get tls.Config by NewTLSConfigFromBaseArgs function
+func RegisterTLSBaseArgs() {
+	flag.StringVar(&tlsCA, "tls_ca", "", "Path to root certificate which will be used with system root certificates to validate peer's certificate")
+	flag.StringVar(&tlsKey, "tls_key", "", "Path to private key that will be used for TLS connections")
+	flag.StringVar(&tlsCert, "tls_cert", "", "Path to certificate")
```

Comment:
Will TLS still work if these options are not specified? I don't remember the story with systems CA certificates lists with Go. @shadinua has seen some issues with that.

Based on `tls_auth` default value we're defaulting to required mutual TLS authentication. All servers *and* clients have to specify their certificates and keys, unless less strict mode is selected.

Reply:
All logic with TLS initialization declared here - https://github.com/cossacklabs/acra/blob/master/network/tls_wrapper.go#L82
We always load system CA files and append tls_ca if it was defined. Plus we allow to use not strict mode of tls without mutual authentication. Then acra-connector should skip tls_key and tls_cert. But if I remember correctly, they are required for AS/AT if tls should work because server should provide any certificate.
Plus this code is not something new, I just copied all args from AS and moved to tls package to reuse it in acra-ee too.

Reply:
Thanks for explanation! 👍